samples: doc: Added README for PM samples

dougwfoster opened the pull request:
Added README.rst for power management **device_pm**
and **latency** samples.

Helps #27805

dougwfoster committed:
samples: doc: Added README for PM samples

Added README.rst for power management device_pm
and latency samples. Also, a pm.rst file was added
to act as a table of contents for PM samples.

Signed-off-by: Doug Foster <dougwfost@gmail.com>

kartben reviewed (changes requested):
This is very nice, thank you! Very minor comments + please add a document similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/portability/portability.rst to act as the table of contents (see the current errors reported in CI giving a hint about this)

thanks again!

dougwfoster committed:
doc: pm: Added labels to PM API doc

Added labels to Device Runtime PM APIs
and System PM APIs in power management
API doc in order to link to different
sections.

Signed-off-by: Doug Foster <dougwfost@gmail.com>

dougwfoster commented:
> This is very nice, thank you! Very minor comments + please add a document similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/portability/portability.rst to act as the table of contents (see the current errors reported in CI giving a hint about this)
> 
> thanks again!

Thanks for the feedback! I made some updates based on your suggestions.

kartben commented on samples/subsys/pm/device_pm/README.rst:
```diff
@@ -0,0 +1,40 @@
+.. zephyr:code-sample:: device-pm
+   :name: Device Power Management
+   :relevant-api: device-runtime-pm-apis
```
This was actually ok! Sorry that this is confusing :)
This, is for augmenting the beginning of each API reference section with cross links to the code sample
The `:ref:`s that needed to be changed are because linking to an actual Doxygen group is actually not easily possible (go figure) so you'd want to link to a manually created anchor instead.
Hope this helps!

dougwfoster replied:
Oh, I see. Thanks for the clarification! I will change these references to what they were before.

kartben commented on samples/subsys/pm/latency/README.rst:
```diff
@@ -0,0 +1,65 @@
+.. zephyr:code-sample:: pm-latency
+   :name: Power Management Latency
+   :relevant-api: system-pm-apis
```
Ditto

dougwfoster commented on samples/subsys/pm/device_pm/README.rst:
```diff
@@ -0,0 +1,40 @@
+********
+
+This sample application uses a dummy driver implementation to demonstrate using the
+:ref:`Device Runtime Power Management API<subsys_pm_device_runtime>` to resume and suspend
```
I added some labels like this in the PM API doc for the link to actually work.

dougwfoster commented:
@kartben Please let me know if you have more suggestions
